packages/react-components/react-table/src/components/DataGrid/useDataGrid.ts

```diff
@@ -45,11 +56,31 @@ export const useDataGrid_unstable = (props: DataGridProps, ref: React.Ref<HTMLEl
       onSelectionChange,
       selectionMode: selectionMode ?? 'multiselect',
     }),
-  ]);
+  ];
+
+  const columnSizingPlugin: TableFeaturePlugin = useTableColumnSizing_unstable({
+    onColumnResize,
+    columnSizingOptions,
+    // The selection cell is not part of the columns, therefore its width needs to be subtracted
+    // from the container to make sure the columns don't overflow the table.
+    containerWidthOffset: selectionMode ? -CELL_WIDTH : 0,
+  });
+
+  if (resizableColumns) {
+    tablePlugins.push(columnSizingPlugin);
+  }
```

Comment:
```suggestion
  if (resizableColumns) {
    tablePlugins.push(columnSizingPlugin);
  }
```

This will break the rule of hooks and cause errors. I updated the resizable columns story and used a button to toggle `resizableColumns`. The plugin is actually a hook and not a pure function.

The way sorting and selection are toggled is by always calling the state and then disabling the interactions... if you can think of a better way that won't violate the order of hooks feel free to try but I couldn't 🤷